DataEngineer/Intermediate/README.md

```diff
@@ -0,0 +1,31 @@
+## Intermediate Data Engineer test
+### Context
+A supermarket stores their transaction data in a relational database for further analysis. The transaction data is uploaded to the system as an CSV file a few times per day. Each transaction has an unique `id`. As a data engineer, your task is to implement and maintain a pipeline to store the transaction data  in the database.
+
+### Elements
+For the test, the following elements are provided:
+- A sqlite database contains historical transactions data
+- A csv file `retail_15_01_2022.csv` contains transaction data of 15/01/2022.
+- The tax for all products is 20%.
+
+### Requirements
+#### Implement the ETL workflow
+Your task is to implement the Python code to store the data from the CSV in the database.
+
+As implementation is only half of the work for a Data Engineer, you're also asked to implement test cases to verify things work as expected. An example of test is provided in the file `test.py`. You need to complete it. Do not hesitate to add more relevant test cases.
+
+#### Explore the data using SQL
+After loading the data in the system, the stakeholder want to extract some information. As a data engineer, you're in charge of the task. Please write the SQL query to answer the following questions:
+- What is the number of transaction on 14/01/2022?
+- What is the total amount, including tax, of all `SELL` transactions?
+- Consider the product `Amazon Echo Dot`, what is the balance (`SELL` - `BUY`) by date?
```

Comment:
```suggestion
- Consider the product `Amazon Echo Dot`, what is the cumulated balance (`SELL` - `BUY`) by date?
```

Reply:
This will change the difficulty of the question. You want the candidate to use Windows function to compute the sliding sums ?

Reply:
> We could add dates of transaction. Dates are always complex to navigate in python.

Yes, the data in the db contains the column transaction_date. I do not included it in the test so the candidate need to parse info from the file name to have it.

Reply:
ok perfect for the dates. cumulated balance make more sense but is not mandatory.

Reply:
I separated into 2 questions. The `cumulated` balance is optional.